### PR TITLE
Controllers tests

### PR DESCRIPTION
**Pull request by BiancaMe**
Add RSpec as a gem to your project.
Create a Request spec file for all your controllers.

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -1,7 +1,45 @@
...
+      get user_path(@user)
+    end
+
+    it 'renders status correct' do
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the show template' do
+      expect(response).to render_template(:show)
+    end
```
Comment (franclobo) [inline spec/requests/users_spec.rb]: - [ ] Please 🙏🏽 correct the test for the correct TDD.

```ruby
Users GET /index renders the index template
     Failure/Error: expect(response).to render_template(:index)
     
     NoMethodError:
       assert_template has been extracted to a gem. To continue using it,
               add `gem 'rails-controller-testing'` to your Gemfile.
     # ./spec/requests/users_spec.rb:20:in `block (3 levels) in <main>'

  4) Users GET /show renders the show template
     Failure/Error: expect(response).to render_template(:show)
     
     NoMethodError:
       assert_template has been extracted to a gem. To continue using it,
               add `gem 'rails-controller-testing'` to your Gemfile.
     # ./spec/requests/users_spec.rb:38:in `block (3 levels) in <main>'
```

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -1,7 +1,46 @@
...
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the show template' do
+      expect(response).to render_template(:show)
+    end
+
+    it 'response body includes correct placeholders text' do
+      expect(response.body).to include('Post selected Info')
+    end
```
Comment (franclobo) [inline spec/requests/posts_spec.rb:44]: - [ ] Please 🙏🏽 correct the test for the correct TDD 

```ruby
1) Posts GET /index renders the index template
     Failure/Error: expect(response).to render_template(:index)
     
     NoMethodError:
       assert_template has been extracted to a gem. To continue using it,
               add `gem 'rails-controller-testing'` to your Gemfile.
     # ./spec/requests/posts_spec.rb:20:in `block (3 levels) in <main>'

  2) Posts GET /show renders the show template
     Failure/Error: expect(response).to render_template(:show)
     
     NoMethodError:
       assert_template has been extracted to a gem. To continue using it,
               add `gem 'rails-controller-testing'` to your Gemfile.
     # ./spec/requests/posts_spec.rb:39:in `block (3 levels) in <main>'
```

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @BiancaMe, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Tests for Controllers ✔️ 
- Correct Git flow ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/users_spec.rb (hunk 1 above)
Comment (BiancaMe) [reply spec/requests/users_spec.rb]: @franclobo Can you help me with the problem? I can't fix the problem when I try to use render_template. 
Can you guide me to solve it? Thank you

File: spec/requests/posts_spec.rb (hunk 2 above)
Comment (BiancaMe) [reply spec/requests/posts_spec.rb:44]: @franclobo  Can you help me with the problem? I can't fix the problem when I try to use render_template.
Can you guide me to solve it? Thank you

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @BiancaMe , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done fixing the failling tests. 🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**